component/src/main/java/io/siddhi/extension/io/file/util/GenerateAppProperties.java

```diff
@@ -0,0 +1,106 @@
+        } else if (Constants.BINARY_FULL.equalsIgnoreCase(mode)) {
+            properties = new HashMap<>();
+            properties.put(Constants.URI, fileURI);
+            properties.put(Constants.READ_FILE_FROM_BEGINNING, Constants.TRUE);
+            properties.put(Constants.ACTION, Constants.READ);
+            properties.put(Constants.POLLING_INTERVAL, fileSourceConfiguration.getFilePollingInterval());
+            properties.put(Constants.FILE_READ_WAIT_TIMEOUT_KEY,
+                    fileSourceConfiguration.getFileReadWaitTimeout());
+            properties.put(Constants.MODE, mode);
+            properties.put(Constants.CRON_EXPRESSION, fileSourceConfiguration.getCronExpression());
+        } else {
+            properties = new HashMap<>();
+            properties.put(Constants.ACTION, Constants.READ);
+            properties.put(Constants.MAX_LINES_PER_POLL, "10");
+            properties.put(Constants.POLLING_INTERVAL, fileSourceConfiguration.getFilePollingInterval());
+            properties.put(Constants.FILE_READ_WAIT_TIMEOUT_KEY,
+                    fileSourceConfiguration.getFileReadWaitTimeout());
+            properties.put(Constants.MODE, mode);
+            properties.put(Constants.HEADER_PRESENT, fileSourceConfiguration.getHeaderPresent());
+            properties.put(Constants.READ_ONLY_HEADER, fileSourceConfiguration.getReadOnlyHeader());
+            properties.put(Constants.CRON_EXPRESSION, fileSourceConfiguration.getCronExpression());
+            properties.put(Constants.URI, fileURI);
+        }
+        return properties;
+    }
+
+    public static Map<String, String> reProcessFileGenerateProperties(FileSourceConfiguration fileSourceConfiguration,
+                                                                      String fileURI, Map<String, String> properties) {
+        String actionAfterProcess = fileSourceConfiguration.getActionAfterProcess();
+        properties.put(Constants.URI, fileURI);
+        properties.put(Constants.ACK_TIME_OUT, "1000");
+        properties.put(Constants.ACTION, actionAfterProcess);
+        return properties;
+    }
+
+    public static String getFileName(String uri, String protocol) {
+        try {
+            URL url = new URL(String.format("%s%s%s", protocol, File.separator, uri));
+            return FilenameUtils.getName(url.getPath());
+        } catch (MalformedURLException e) {
+            log.error(String.format("Failed to extract file name from the uri '%s '.", uri), e);
+            return null;
+        }
+    }
+
+    public static String constructPath(String baseUri, String fileName) {
+        if (baseUri != null && fileName != null) {
+            if (baseUri.endsWith(File.separator)) {
+                return String.format("%s%s", baseUri, fileName);
+            } else {
+                return String.format("%s%s%s", baseUri, File.separator, fileName);
+            }
+        } else {
+            return null;
+        }
+    }
```

Comment:
Can we add these two methods to the existing [Util.java](https://github.com/siddhi-io/siddhi-io-file/blob/bea46527dba085b2c5bcc206f49786b72192804e/component/src/main/java/io/siddhi/extension/io/file/util/Util.java#L30) class?

Reply:
fixed in a6f1f669af7b573c96bc9d004d0244e4dc9c7e5c